Framework/Kernel/inc/MantidKernel/IPropertyManager.h

```diff
@@ -180,6 +180,9 @@ class MANTID_KERNEL_DLL IPropertyManager {
    *  @throw std::invalid_argument  if the name argument is empty
    */
   void declareProperty(const std::string &name, const char *value, const unsigned int direction) {
+    if (value == nullptr)
+      throw std::invalid_argument("Attempted to set " + name + " to nullptr");
```

Comment:
This seems a sensible check but I'm just wondering why it was needed?

Reply:
It was to fix a cpp compiler warning that says `value` could be nullptr. I don't why this appears right now. I can reverse it and add a warning suppression if it is better

Reply:
Ok, thanks - no need to change it, I was just curious :)